components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/jwt/JWKSBasedJWTValidator.java

```diff
@@ -156,13 +163,23 @@ private void checkCertificateValidity(String jwksUri, SignedJWT jwt) throws Malf
             }
 
             if (x509Certificate != null) {
+
+                String alg = jwt.getHeader().getAlgorithm().getName();
```

Comment:
Is there any possibility to throw NP from here?

Reply:
No. At the upper level, we validate the header and algorithm during the setting process. If either is null, an exception is thrown.